goaviatrix/firenet.go

```diff
@@ -365,3 +367,36 @@ func (c *Client) EditFireNetEgress(fireNet *FireNet) error {
 	}
 	return nil
 }
+
+func (c *Client) EditFireNetHashingAlgorithm(fireNet *FireNet) error {
+	Url, err := url.Parse(c.baseURL)
+	if err != nil {
+		return errors.New(("url Parsing failed for edit_firenet") + err.Error())
+	}
+	editFireNet := url.Values{}
+	editFireNet.Add("CID", c.CID)
+	editFireNet.Add("action", "edit_firenet")
+	editFireNet.Add("vpc_id", fireNet.VpcID)
+	editFireNet.Add("firewall_hashing", fireNet.HashingAlgorithm)
+
+	Url.RawQuery = editFireNet.Encode()
+	resp, err := c.Get(Url.String(), nil)
+	if err != nil {
```

Comment:
This can be simplified with the PostAPI function. This way we are not repeating the same JSON decoding logic

```
func (c *Client) EditFireNetHashingAlgorithm(fireNet *FireNet) error {
	action := "edit_firenet"
	data := map[string]string{
		"action":           "edit_firenet",
		"CID":              c.CID,
		"vpc_id":           fireNet.VpcID,
		"firewall_hashing": fireNet.HashingAlgorithm,
	}
	checkFunc := func(act, reason string, ret bool) error {
		if !ret {
			if strings.Contains(reason, "configuration not changed") {
				return nil
			}
			return fmt.Errorf("rest API %s Post failed: %s", action, reason)
		}
		return nil
	}

	return c.PostAPI(action, data, checkFunc)
}
```

Reply:
done